### ISSUE TITLE
Running HDBSCAN on data with duplicates

### ISSUE THREAD

**Issue by azizkayumov** [petabi/petal-clustering#70]
HDBSCAN's flat clustering results rely on cluster stability calculation which is prone to output erroneous clustering partitions if there are [duplicate data objects](https://github.com/scikit-learn-contrib/hdbscan/issues/94).

This current implementation handles [this case as follows](https://github.com/petabi/petal-clustering/blob/9d2c1ca0d35033487674cc97f07b94e1c67acc8d/src/hdbscan.rs#L216):
```
        let info = mst[node - n];
        let lambda = if info.2 > A::zero() {
            A::one() / info.2
        } else {
            A::max_value()
        };
```
Python HDBSCAN uses `infinity` [instead](https://github.com/scikit-learn-contrib/hdbscan/blob/c201b2e75d42fd0fb7f3aed6b0992afa3c5501c7/hdbscan/_hdbscan_tree.pyx#L106):
```
        children = hierarchy[node - num_points]
        left = <np.intp_t> children[0]
        right = <np.intp_t> children[1]
        if children[2] > 0.0:
            lambda_value = 1.0 / children[2]
        else:
            lambda_value = INFTY
```

Basically, this behavior depends on the parameter selection for `minPts`, as long as any core distance is not zero, flat clustering results should not be affected. 

Do you think it would be a good idea to warn the users about this behavior? The original Java implementation of HDBSCAN leaves a [warning message](https://github.com/antoniocavalcante/hdbscan-java/blob/57caee3e080d0e69d3bf5dc7b7872cedae58b244/src/ca/ualberta/cs/hdbscanstar/HDBSCANStar.java#L41), advising that the user should increase his `minPts`. Somehow the Python version is silent about this (so is this Rust version), which may leave the users believe in the flat clustering results and wrongly confuse them to search for other alternatives.
